modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppQt5QHttpEngineServerCodegen.java

```diff
@@ -431,4 +443,71 @@ public String toParamName(String name) {
     public String getTypeDeclaration(String str) {
         return str;
     }
+      
+    @Override
+    protected boolean needToImport(String type) {
+        return StringUtils.isNotBlank(type) && !defaultIncludes.contains(type)
+                && !nonFrameworkPrimitives.contains(type);
+    }
+    
+    @Override
+    @SuppressWarnings("unchecked")
+    public Map<String, Object> postProcessOperations(Map<String, Object> objs) {
+        Map<String, Object> objectMap = (Map<String, Object>) objs.get("operations");
+        List<CodegenOperation> operations = (List<CodegenOperation>) objectMap.get("operation");
+        
+        List<Map<String, String>> imports = (List<Map<String, String>>) objs.get("imports");
+        for (CodegenOperation operation : operations) {
+            if (operation.returnBaseType != null && needToImport(operation.returnBaseType)) {
+                if(!isIncluded(operation.returnBaseType, imports)) {
+                    imports.add(createMapping("import", operation.returnBaseType));
+                }
+            }
+            List<CodegenParameter> params = new ArrayList<CodegenParameter>();
+            if (operation.bodyParam != null)params.add(operation.bodyParam);
+            if (operation.allParams != null)params.addAll(operation.allParams);
```

Comment:
If you add `allParams` and also add query/body/header/query parameters, will `params` contain duplicated parameters?

I would also suggest adding a few comments to different code block in this function (postProcessOperations) so as to explain a bit more about the purpose.

Reply:
Ahhh ... does `allParams` include everything?
But actually depending on param type I need to check data type.
I will add comments. The goal here is that I consider Qt5 handled data types as native but I need to add the header, whereas default behavior is to not add headers for native types. There is no special handling needed for Qt5 handled types because the framework does everything.

Reply:
Yup https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L2344-L2426

Search for `allParams` in that code block.

Reply:
Thanks, I have removed the duplicated parameters and added comments